internal/controllers/machinedeployment/mdutil/util_test.go

```diff
@@ -755,6 +765,151 @@ func TestAnnotationUtils(t *testing.T) {
 	})
 }
 
+func TestComputeMachineSetAnnotations(t *testing.T) {
+	deployment := generateDeployment("nginx")
+	deployment.Spec.Replicas = pointer.Int32(3)
+	maxSurge := intstr.FromInt(1)
+	maxUnavailable := intstr.FromInt(0)
+	deployment.Spec.Strategy = &clusterv1.MachineDeploymentStrategy{
+		Type: clusterv1.RollingUpdateMachineDeploymentStrategyType,
+		RollingUpdate: &clusterv1.MachineRollingUpdateDeployment{
+			MaxSurge:       &maxSurge,
+			MaxUnavailable: &maxUnavailable,
+		},
+	}
+	deployment.Annotations = map[string]string{
+		corev1.LastAppliedConfigAnnotation: "last-applied-configuration",
+		"key1":                             "value1",
+	}
+
+	tests := []struct {
+		name       string
+		deployment *clusterv1.MachineDeployment
+		oldMSs     []*clusterv1.MachineSet
+		ms         *clusterv1.MachineSet
+		want       map[string]string
+		wantErr    bool
+	}{
+		{
```

Comment:
Kudos, nice test!

Was really easily understandable what the expected behavior is for me.